frontend/src/pages/Admin/InventoryManagement.js

```diff
@@ -42,6 +45,42 @@ const InventoryManagement = () => {
     refreshFlag,
   )
 
+  // Filter products based on search query and dropdown filter
+  useEffect(() => {
+    if (productData.body) {
+      let results = [...productData.body];
+      
+      // Apply text search filter
+      if (searchQuery) {
+        const query = searchQuery.toLowerCase();
+        results = results.filter(product => 
+          product.itemName?.toLowerCase().includes(query) || 
+          product.description?.toLowerCase().includes(query) || 
+          product.category?.toLowerCase().includes(query) ||
```

Comment:
I was thinking if we could add a separate `Select` to select the category. I like that you also considered the description when searching.